packages/plugin-patch/sources/patchUtils.ts

```diff
@@ -172,3 +172,18 @@ export async function extractPackageToDisk(locator: Locator, {cache, project}: {
   xfs.detachTemp(temp);
   return temp;
 }
+
+export async function diffFolders(folderA: PortablePath, folderB: PortablePath) {
+  const folderAN = npath.fromPortablePath(folderA).replace(/\\/g, `/`);
+  const folderBN = npath.fromPortablePath(folderB).replace(/\\/g, `/`);
+
+  const {stdout} = await execUtils.execvp(`git`, [`diff`, `--ignore-cr-at-eol`, `--full-index`, `--no-index`, folderAN, folderBN], {
+    cwd:npath.toPortablePath(process.cwd()),
+  });
+
+  return stdout
+    .replace(new RegExp(`(a|b)${miscUtils.escapeRegExp(`/${folderAN.substr(folderAN.startsWith(`/`) ? 1 : 0)}/`)}`, `g`), `$1/`)
```

Comment:
Why cover `(a|b)`, since the replacement is on `a/folderAN`? Even if `folderAN === folderBN`, it'll be replaced at the following replacement pass, right?

Reply:
In a "delete" change `folderAN`/`folderBN` is both `a/` and `b/`